File: 10-facade/10-facade.php
<?php

class Audio
{
    public function init(): void
    {
        echo "audio initialized\n";
    }
}

class Video
{
    public function init(): void
    {
        echo "video initialized\n";
    }
}

class MediaFacade
{
    public function __construct(private Audio $audio, private Video $video)
    {
    }
    public function start()
    {
        $this->audio->init();
        $this->video->init();
    }
}

$audio = new Audio();
$video = new Video();
$media = new MediaFacade($audio, $video);
$media->start();

File: 10-facade/10-facade.go
package main

import "fmt"

type Audio struct{}

func (a Audio) init() string {
	return "audio on"
}

type Video struct{}

func (v Video) init() string {
	return "video on"
}

type Facade struct {
	audio Audio
	video Video
}

func (f Facade) facade() {
	fmt.Println(f.audio.init())
	fmt.Println(f.video.init())
}

func main() {
	a := Audio{}
	v := Video{}
	f := Facade{audio: a, video: v}

	f.facade()
}
